pkg/yurtmanager/controller/yurtappoverrider/yurtappoverrider_controller.go

```diff
@@ -128,8 +153,35 @@ func (r *ReconcileYurtAppOverrider) Reconcile(_ context.Context, request reconci
 		return reconcile.Result{}, nil
 	}
 
-	deployments := v1.DeploymentList{}
+	if cacheOverrider, ok := r.CacheOverriderMap[instance.Name]; ok {
+		if reflect.DeepEqual(cacheOverrider.Entries, instance.Entries) {
+			return reconcile.Result{}, nil
+		}
+	} else {
+		r.CacheOverriderMap[instance.Name] = instance.DeepCopy()
+	}
 
+	switch instance.Subject.Kind {
+	case "YurtAppSet":
+		appSet := &appsv1alpha1.YurtAppSet{}
+		if err := r.Get(context.TODO(), client.ObjectKey{Namespace: instance.Namespace, Name: instance.Subject.Name}, appSet); err != nil {
+			return reconcile.Result{}, err
+		}
+		if appSet.Spec.WorkloadTemplate.StatefulSetTemplate != nil {
+			r.recorder.Event(instance.DeepCopy(), corev1.EventTypeWarning, fmt.Sprintf("unable to override statefulset workload of %s", appSet.Name), "It is not supported to overrider statefulset now")
```

Comment:
please add `return reconcile.Result{}, nil` for short-circuit return.

Reply:
fix